### PR TITLE
Move ImmutableOpenMap usage to j.u.Map

### PR DESCRIPTION
**Pull request by junqiu-lei**
### Description
* Backport part of https://github.com/opensearch-project/k-NN/pull/885 to 2.x
* Fix run error from "org.opensearch.core.common" import
 
### Issues Resolved
[List any issues this PR will resolve]
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed as per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/k-NN/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by vamshin — APPROVED** [review]
LGTM! Thanks

**Review by vamshin — APPROVED** [review]
LGTM! Thanks

**Comment by junqiu-lei** [comment]
CI failed:
```
> Task :integTest
OpenJDK 64-Bit Server VM warning: Ignoring option --illegal-access=warn; support was removed in 17.0

REPRODUCE WITH: ./gradlew ':integTest' --tests "org.opensearch.knn.index.FaissIT.testEndToEnd_fromModel" -Dtests.seed=BA64F5A0A47B5FD5 -Dtests.security.manager=false -Dtests.locale=sl -Dtests.timezone=GMT -Druntime.java=17

org.opensearch.knn.index.FaissIT > testEndToEnd_fromModel FAILED
    org.opensearch.client.ResponseException: method [GET], host [http://[::1]:46745], URI [/_plugins/_knn/models/test-model], status line [HTTP/1.1 400 Bad Request]
    {"error":{"root_cause":[{"type":"illegal_argument_exception","reason":"model ID cannot be empty"}],"type":"illegal_argument_exception","reason":"model ID cannot be empty"},"status":400}
        at app//org.opensearch.client.RestClient.convertResponse(RestClient.java:375)
        at app//org.opensearch.client.RestClient.performRequest(RestClient.java:345)
        at app//org.opensearch.client.RestClient.performRequest(RestClient.java:320)
        at app//org.opensearch.knn.KNNRestTestCase.getModel(KNNRestTestCase.java:1131)
        at app//org.opensearch.knn.KNNRestTestCase.assertTrainingSucceeds(KNNRestTestCase.java:1143)
        at app//org.opensearch.knn.index.FaissIT.testEndToEnd_fromModel(FaissIT.java:243)

    org.opensearch.client.ResponseException: method [GET], host [http://[::1]:46745], URI [/_plugins/_knn/models/test-model], status line [HTTP/1.1 400 Bad Request]
    {"error":{"root_cause":[{"type":"illegal_argument_exception","reason":"model ID cannot be empty"}],"type":"illegal_argument_exception","reason":"model ID cannot be empty"},"status":400}
        at app//org.opensearch.client.RestClient.convertResponse(RestClient.java:375)
        at app//org.opensearch.client.RestClient.performRequest(RestClient.java:345)
        at app//org.opensearch.client.RestClient.performRequest(RestClient.java:320)
        at app//org.opensearch.knn.ODFERestTestCase.deleteModels(ODFERestTestCase.java:223)
        at app//org.opensearch.knn.ODFERestTestCase.wipeIndexContent(ODFERestTestCase.java:201)
        at app//org.opensearch.knn.ODFERestTestCase.wipeAllODFEIndices(ODFERestTestCase.java:186)
        at java.base@17.0.7/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
```

**Comment by navneet1v** [comment]
@junqiu-lei please check why this test: `Build and Test k-NN / Build and Test k-NN Plugin (17, ubuntu-latest) (pull_request) ` is failing

File: src/main/java/org/opensearch/knn/plugin/rest/RestGetModelHandler.java [hunk 1]
```diff
@@ -50,7 +50,7 @@ public List<Route> routes() {
     @Override
     protected RestChannelConsumer prepareRequest(RestRequest restRequest, NodeClient client) throws IOException {
         String modelID = restRequest.param(MODEL_ID);
-        if (!Strings.hasText(modelID)) {
+        if (!StringUtils.isBlank(modelID)) {
```
Comment (navneet1v) [inline src/main/java/org/opensearch/knn/plugin/rest/RestGetModelHandler.java:53]: if we are replacing Strings.hasTest, we should replace it with StringUtils.isNotBlank

File: src/main/java/org/opensearch/knn/plugin/rest/RestDeleteModelHandler.java [hunk 2]
```diff
@@ -58,7 +58,7 @@ public List<Route> routes() {
     @Override
     protected RestChannelConsumer prepareRequest(RestRequest request, NodeClient client) {
         String modelID = request.param(MODEL_ID);
-        if (!Strings.hasText(modelID)) {
+        if (StringUtils.isNotBlank(modelID)) {
```
Comment (navneet1v) [inline src/main/java/org/opensearch/knn/plugin/rest/RestDeleteModelHandler.java:61]:
```suggestion
        if (StringUtils.isBlank(modelID)) {
```

File: src/main/java/org/opensearch/knn/plugin/rest/RestGetModelHandler.java (hunk 1 above)
Comment (junqiu-lei) [reply src/main/java/org/opensearch/knn/plugin/rest/RestGetModelHandler.java]: updated

**Comment by junqiu-lei** [comment]
CI build [Build and Test k-NN Plugin (17, ubuntu-latest)](https://github.com/opensearch-project/k-NN/actions/runs/5495538778/jobs/10015082597?pr=955#logs) passed. @navneet1v

**Comment by junqiu-lei** [comment]
The [Build and Test k-NN Plugin (11, macos-latest)](https://github.com/opensearch-project/k-NN/actions/runs/5495538778/jobs/10024638433#logs) failed error come from opensearch-2.9.0-SNAPSHOT.jar:2.9.0-SNAPSHOT:
```
» ERROR][o.o.b.OpenSearchUncaughtExceptionHandler] [integTest-0] fatal error in thread [opensearch[integTest-0][generic][T#1]], exiting
»  java.lang.AbstractMethodError: Receiver class org.opensearch.knn.plugin.KNNPlugin$$Lambda$2050/0x0000000800886c40 does not define or inherit an implementation of the resolved method 'abstract java.lang.Object read(org.opensearch.core.common.io.stream.BaseStreamInput)' of interface org.opensearch.core.common.io.stream.BaseWriteable$Reader.
»  	at org.opensearch.common.io.stream.NamedWriteableAwareStreamInput.readNamedWriteable(NamedWriteableAwareStreamInput.java:63) ~[opensearch-2.9.0-SNAPSHOT.jar:2.9.0-SNAPSHOT]
»  	at org.opensearch.cluster.NamedDiffableValueSerializer.read(NamedDiffableValueSerializer.java:55) ~[opensearch-2.9.0-SNAPSHOT.jar:2.9.0-SNAPSHOT]
»  	at org.opensearch.cluster.NamedDiffableValueSerializer.read(NamedDiffableValueSerializer.java:45) ~[opensearch-2.9.0-SNAPSHOT.jar:2.9.0-SNAPSHOT]
»  	at org.opensearch.cluster.DiffableUtils$MapDiff.<init>(DiffableUtils.java:239) ~[opensearch-2.9.0-SNAPSHOT.jar:2.9.0-SNAPSHOT]
»  	at org.opensearch.cluster.DiffableUtils$JdkMapDiff.<init>(DiffableUtils.java:137) ~[opensearch-2.9.0-SNAPSHOT.jar:2.9.0-SNAPSHOT]
»  	at org.opensearch.cluster.DiffableUtils.readJdkMapDiff(DiffableUtils.java:112) ~[opensearch-2.9.0-SNAPSHOT.jar:2.9.0-SNAPSHOT]
»  	at org.opensearch.cluster.metadata.Metadata$MetadataDiff.<init>(Metadata.java:1017) ~[opensearch-2.9.0-SNAPSHOT.jar:2.9.0-SNAPSHOT]
»  	at org.opensearch.cluster.metadata.Metadata.readDiffFrom(Metadata.java:948) ~[opensearch-2.9.0-SNAPSHOT.jar:2.9.0-SNAPSHOT]
»  	at org.opensearch.cluster.ClusterState$ClusterStateDiff.<init>(ClusterState.java:845) ~[opensearch-2.9.0-SNAPSHOT.jar:2.9.0-SNAPSHOT]
»  	at org.opensearch.cluster.ClusterState.readDiffFrom(ClusterState.java:752) ~[opensearch-2.9.0-SNAPSHOT.jar:2.9.0-SNAPSHOT]
»  	at org.opensearch.cluster.coordination.PublicationTransportHandler.handleIncomingPublishRequest(PublicationTransportHandler.java:189) ~[opensearch-2.9.0-SNAPSHOT.jar:2.9.0-SNAPSHOT]
»  	at org.opensearch.cluster.coordination.PublicationTransportHandler.lambda$new$0(PublicationTransportHandler.java:117) ~[opensearch-2.9.0-SNAPSHOT.jar:2.9.0-SNAPSHOT]
»  	at org.opensearch.transport.RequestHandlerRegistry.processMessageReceived(RequestHandlerRegistry.java:106) ~[opensearch-2.9.0-SNAPSHOT.jar:2.9.0-SNAPSHOT]
»  	at org.opensearch.transport.TransportService$8.doRun(TransportService.java:1091) ~[opensearch-2.9.0-SNAPSHOT.jar:2.9.0-SNAPSHOT]
»  	at org.opensearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:854) ~[opensearch-2.9.0-SNAPSHOT.jar:2.9.0-SNAPSHOT]
»  	at org.opensearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:52) ~[opensearch-2.9.0-SNAPSHOT.jar:2.9.0-SNAPSHOT]
»  	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) ~[?:?]
»  	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) ~[?:?]
»  	at java.lang.Thread.run(Thread.java:829) [?:?]
»   ↓ last 40 non error or warning messages from /Users/runner/work/k-NN/k-NN/build/testclusters/integTest-0/logs/opensearch.stdout.log ↓
```

**Review by navneet1v — APPROVED** [review]

**Review by martin-gaievski — APPROVED** [review]

**Pull request merged**